src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallGlobalOrToolPathCommand.cs

```diff
@@ -114,66 +115,76 @@ public override int Execute()
+                            packageId: _packageId,
+                            versionRange: versionRange,
+                            targetFramework: _framework,
+                            verbosity: _verbosity,
+                            isGlobalTool: true
+                        );
+
+                        EnsureVersionIsHigher(oldPackageNullable, newInstalledPackage, _allowPackageDowngrade);
+
+                        NuGetFramework framework;
+                        if (string.IsNullOrEmpty(_framework) && newInstalledPackage.Frameworks.Count() > 0)
+                        {
+                            framework = newInstalledPackage.Frameworks
+                                .Where(f => f.Version < (new NuGetVersion(Product.Version)).Version)
+                                .MaxBy(f => f.Version);
+                        }
+                        else
+                        {
+                            framework = string.IsNullOrEmpty(_framework) ?
+                                null :
+                                NuGetFramework.Parse(_framework);
+                        }
+                        string appHostSourceDirectory = _shellShimTemplateFinder.ResolveAppHostSourceDirectoryAsync(_architectureOption, framework, RuntimeInformation.ProcessArchitecture).Result;
 
-                    foreach (string w in newInstalledPackage.Warnings)
-                    {
-                        _reporter.WriteLine(w.Yellow());
-                    }
-                    if (_global)
-                    {
-                        _environmentPathInstruction.PrintAddPathInstructionIfPathDoesNotExist();
-                    }
+                        foreach (RestoredCommand command in newInstalledPackage.Commands)
+                        {
+                            shellShimRepository.CreateShim(command.Executable, command.Name, newInstalledPackage.PackagedShims);
+                        }
+
+                        foreach (string w in newInstalledPackage.Warnings)
+                        {
+                            _reporter.WriteLine(w.Yellow());
+                        }
+                        if (_global)
+                        {
+                            _environmentPathInstruction.PrintAddPathInstructionIfPathDoesNotExist();
+                        }
 
-                    PrintSuccessMessage(oldPackageNullable, newInstalledPackage);
-                });
+                        PrintSuccessMessage(oldPackageNullable, newInstalledPackage);
+                    });
 
-                scope.Complete();
-            }
+                    scope.Complete();
+                }
+                catch (Exception)
+                {
+                    // Log the exception or perform other error handling if needed.
+                    // Rollback the transaction to ensure consistency.
+                    scope.Dispose();
+                    throw;
+                }
```

Comment:
Is this actually needed?  The `using` block should ensure that the scope is disposed when it exits that block, whether or not it happens via an exception.